File: services/upx/Entities/Transaction.cs
using Foundation.Common.Entities;

namespace Foundation.Services.UPx;

public enum TransactionType
{
    Claim,
    Spend
}

public class Transaction : BaseEntity
{
    public required TransactionType TransactionType { get; set; }

    public required string UserId { get; set; } = default!;

    public string ClaimId { get; set; } = default!;

    public required double Credits { get; set; }

    public required long Timestamp { get; set; }

    public string Description { get; set; } = default!;
}

File: services/echo/internal/structures/transaction.go
package structures

type TransactionType int

const (
	CLAIM TransactionType = iota
	SPEND
)

type Transaction struct {
	TransactionType TransactionType `json:"transaction_type"`
	UserId          string          `json:"user_id"`
	ClaimId         string          `json:"claim_id"`
	Credits         float32         `json:"credits"`
	Timestamp       int64           `json:"timestamp"`
	Description     string          `json:"description"`
}
